File: src/community/script/py/src/test/resources/org/geoserver/script/py/buffer-multipleOutputs.py
from geoserver.wps import process
from geoscript.geom import Geometry

@process(
  inputs={'geom': (Geometry, 'The geometry to buffer'), 'distance':(float,'The buffer distance')}, 
  outputs={'geom': (Geometry, 'The buffered geometry'), 'distance':(float,'The buffer distance')}, 
  title='Buffer', 
  description='Buffers a geometry')
def run(geom, distance):
  return {'geom': geom.buffer(distance), 'distance': distance}

File: src/community/script/rb/src/test/resources/org/geoserver/script/rb/buffer-multipleOutputs.rb
require 'java'

$title = 'Buffer'
$description = 'Buffers a geometry'

$inputs = { 
  "geom" => { 
    "title" => "The geometry to buffer", 
    "type" => org.locationtech.jts.geom.Geometry.java_class
   },
  "distance" => { 
    "title" => "The buffer distance", 
    "type" => java.lang.Double.java_class
  }
} 

$outputs = {
  "geom" => { 
    "title" => "The geometry to buffer", 
    "type" => org.locationtech.jts.geom.Geometry.java_class
   },
  "distance" => { 
    "title" => "The buffer distance", 
    "type" => java.lang.Double.java_class
  }
}

def run(inputs)
  return { "geom" => inputs["geom"].buffer(inputs["distance"]), "distance" => inputs["distance"] }
  
end